canonical-web-and-design/maas-ui
# feat(ui): update KVM compose form

## Done

- Updated KVM compose form to include hugepages backing and pinning specific cores. Note that I haven't includes the available cores helper text because I'll be changing over a lot of the NUMA-aware data to the new structure very soon.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to a KVM details page and open the compose form.
- Check the hugepages checkbox and choose the pin cores radio button.
- Check that the pinned cores ranges validation works as expected
- Submit the form and check that a new VM is added to the pod in state with the correct data: `pod.resources.vms`

## Fixes

Fixes #2237 

## Screenshot

![Screenshot_2021-03-09 LXD project default bolla MAAS](https://user-images.githubusercontent.com/25733845/110439320-0c872300-8103-11eb-90d4-365b048df1d6.png)
